spec/models/source_spec.rb

```diff
@@ -10,6 +10,7 @@
   before do
     allow(terra).to receive(:new).and_return(instance_terra)
     allow(instance_terra).to receive(:validate)
+    allow_any_instance_of(SourceValidator).to receive(:terraform).and_return(instance_terra)
```

Comment:
seems pretty magic this thing :grin:

Reply:
This is the RSpec API for configuring stubs: here we stub the private `terraform` method on `SourceValidator` instances, so that it internally uses a double of the `Terraform` service, instead of the real one.

Reply:
yep was just comment of minor importance no worries